packages/core/test/discovery.test.js

```diff
@@ -2430,6 +2430,29 @@ describe('Discovery', () => {
     });
   });
 
+  describe('discovery parameters =>', () => {
+    it('checks for waitForTimeout and waitForSelector at discovery', async () => {
```

Comment:
`describe('waitForSelector/waitForTimeout')`

   describe('when Js is enabled')
     it('waits for Timeout')
     it('waits for selector')
 
  describe('when JS is disabled')
    it('doesn't waits for Timeout')

- Add this test case as well
- Also spyOn(page), and validate that page.eval was called when JS in enabled